datafusion/src/datasource/datasource.rs

```diff
@@ -102,6 +102,7 @@ pub trait TableProvider: Sync + Send {
         // If set, it contains the amount of rows needed by the `LogicalPlan`,
         // The datasource should return *at least* this number of rows if available.
         limit: Option<usize>,
+        max_partitions: usize,
```

Comment:
What about calling it `target_partitions` ? In the future we might add support for (column-based) partitioning in datasources, in that scenario you might end up with more than `max_partitions`

Reply:
I agree.  fixed